tests/e2e/commands/blockchain/blockchain_command.go

```diff
@@ -0,0 +1,111 @@
+package blockchain
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/ava-labs/avalanche-cli/tests/e2e/utils"
+)
+
+// TestCase represents a single test case configuration
+type TestCase struct {
+	Name          string            `json:"name"`
+	Flags         map[string]string `json:"flags"`
+	ExpectedError string            `json:"expectedError,omitempty"`
+}
+
+// TestJSONConfig represents the json configuration that contains cli command flag inputs
+type TestJSONConfig struct {
+	GlobalFlags  map[string]interface{} `json:"globalFlags"`
+	HappyPath    []TestCase             `json:"happyPath"`
+	NotHappyPath []TestCase             `json:"notHappyPath"`
+}
+
+var avalancheBinaryPath = "./bin/avalanche"
+
+// SetAvalancheBinaryPath sets the path to the avalanche binary
+func SetAvalancheBinaryPath(path string) {
+	avalancheBinaryPath = path
+}
+
+// TestCommandWithJSONConfig tests a CLI command with flag inputs from a JSON file
+func TestCommandWithJSONConfig(command string, configPath string, testCase *TestCase) (string, error) {
```

Comment:
love the idea - any plan to move it to some test common directories to enable it reusable by all other commands?

Reply:
Thanks! Yes i will move it to external directories

Reply:
Made it general